plugins/rbac/src/hooks/useRoles.ts

```diff
@@ -70,42 +77,79 @@ export const useRoles = (
     permission: policyEntityUpdatePermission,
     resourceRef: policyEntityUpdatePermission.resourceType,
   });
+
+  React.useEffect(() => {
+    const conditionsPromises: Promise<RoleWithConditionalPoliciesCount>[] =
+      (Array.isArray(roles) &&
+        roles.map(async role => {
+          const conditionalPolicies = await rbacApi.getRoleConditions(
+            role.name,
+          );
+
+          return {
+            ...role,
+            conditionalPoliciesCount: Array.isArray(conditionalPolicies)
+              ? conditionalPolicies.length
+              : 0,
+          };
+        })) ||
+      [];
+    const fetchAllPermissionPolicies = async () => {
+      if (Array.isArray(roles)) {
+        try {
+          const updatedRoles: RoleWithConditionalPoliciesCount[] =
+            await Promise.all(conditionsPromises);
```

Comment:
FYI Promise.all will never throw an error because of how the api calls are handled above as in case one of the api calls fail there will always still be an object just the conditionalPoliciesCount will be 0. It should be okay to keep the count as 0 for the api call which failed but the current code is not correct as try and catch will not be used here as the code expects. Also if we keep the count as 0 user will not know what happened so ideally user should know for which role this api call failed and still calculate the count for others who passed. Also I think you should not make the roles data generation dependent upon this function as for example if the code actually throws an error once you fix it the roles will never set and hence even the roles data will not be generated.

Reply:
@divyanshiGupta What is the best way to handle any role condition fetching failures? I can still return the role but if the condition fetching failed, do we need to let the user know at meanwhile show the role data in the list just missing condition data?

Reply:
So if you see here https://github.com/janus-idp/backstage-plugins/blob/b83a315113ef69c5a85f0a100d730a99a4590b48/plugins/rbac/src/hooks/useRoles.ts#L172 error is being set. You can have another param for conditions-error
If you follow this:
```
conditionsPromises = roles.map(role =>  rbacApi.getRoleConditions(role.name));
const conditions = await Promise.all(conditionsPromises);
```
the conditions will be an array of RoleConditionalPolicyDecision<PermissionAction>[] | Response. You will need extract out the Responses and use its statusText for the error message.
```
const err = conditions
            .map(c => (c as unknown as Response)?.statusText)
            .filter(m => m);

          if (err.length > 0) {
            throw new Error(
              `Unable to retrieve conditions. ${err.join('\n')}`,
            );
          }
```
And set the error in state in catch and use it here https://github.com/janus-idp/backstage-plugins/blob/b83a315113ef69c5a85f0a100d730a99a4590b48/plugins/rbac/src/hooks/useRoles.ts#L172. (edited)

Reply:
I have added an alert warning when some role condition fetching fails. I made it a warning message as it won't block roles data listing. Let me know how do you think.
![image](https://github.com/janus-idp/backstage-plugins/assets/26255262/3c7b1684-1ab7-4796-b5eb-c0d58dafcf42)